lib/pool/BaseConnectionPool.js

```diff
@@ -243,7 +243,6 @@ class BaseConnectionPool {
         url: new URL(address),
         id: ids[i],
         roles: Object.assign({
-          [Connection.roles.MASTER]: false,
```

Comment:
is this ok to remove?

Reply:
I think you could. My thinking process is below. I set `master` and `cluster_manager` both optional in node roles. For this `nodesToHost` method, I think it transfers the nodes objects to a hosts object (hosts array). 

```
  nodesToHost (nodes, protocol) {
    const ids = Object.keys(nodes)
    const hosts = []

    for (let i = 0, len = ids.length; i < len; i++) {
      const node = nodes[ids[i]]
      let address = node.http.publish_address
      const parts = address.split('/')
      // the url is in the form of hostname/ip:port
      if (parts.length > 1) {
        const hostname = parts[0]
        const port = parts[1].match(/((?::))(?:[0-9]+)$/g)[0].slice(1)
        address = `${hostname}:${port}`
      }

      address = address.slice(0, 4) === 'http'
        /* istanbul ignore next */
        ? address
        : `${protocol}//${address}`
      const roles = node.roles.reduce((acc, role) => {
        acc[role] = true
        return acc
      }, {})

      hosts.push({
        url: new URL(address),
        id: ids[i],
        roles: Object.assign({
          [Connection.roles.DATA]: false,
          [Connection.roles.INGEST]: false
        }, roles)
      })
    }

    return hosts
  }
```

For customers who use `master`, their hosts would be one/several nodes with `master = true` and other nodes with `data` role or `ingest` role. When OpenSearch tests whether it is a master node [here](https://github.com/opensearch-project/OpenSearch/blob/ea3148312622cd1362593e3623d498e57913ae45/client/rest/src/main/java/org/opensearch/client/Node.java#L216) , it should work fine. For new customers, when they pass `cluster_manager`, OpenSearch could also find cluster managers. As [in this test](https://github.com/opensearch-project/opensearch-js/blob/238ce88300e01e875513a50ff15cda146a8a022a/test/unit/connection-pool.test.js#L456) , though it might be not very realistic because customers might not pass both `cluster_manager` and `master`, we could see node roles are correctly marked. 

Rocky, do you have any concerns or observe any breaking changes here? or better ideas?

Reply:
To sum up you mean that this stuff is just used within this repo and you handle it the logic of ensuring the role is mapped correctly?